Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Physics;
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// Tap to place is a far interaction component used to place objects on a surface.
+    /// </summary>
+    public class TapToPlace : Solver, IMixedRealityPointerHandler
+    {
+        [Experimental]
+        [SerializeField]
+        [Tooltip("The game object that will be placed if it is selected")]
+        private GameObject gameObjectToPlace;
+
+        /// <summary>
+        /// The game object that will be placed if it is selected
+        /// </summary>
+        public GameObject GameObjectToPlace
+        {
+            get => gameObjectToPlace;
+            set
+            {
+                if (value != null && gameObjectToPlace != value)
+                {
+                    gameObjectToPlace = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Check if a collider is present on the GameObjectToPlace
+        /// </summary>
+        public bool ColliderPresent
+        {
+            get
+            {
+                if (GameObjectToPlace.GetComponent<Collider>() == null)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("The default distance (in meters) an object will be placed relative to the TrackedTargetType forward in the SolverHandler." +
+            "The GameObjectToPlace will be placed at the default placement distance if a surface is not hit by the raycast.")]
+        private float defaultPlacementDistance = 1.5f;
+
+        /// <summary>
+        /// The default distance (in meters) an object will be placed relative to the TrackedTargetType forward in the SolverHandler.
+        /// The GameObjectToPlace will be placed at the default placement distance if a surface is not hit by the raycast.
+        /// </summary>
+        public float DefaultPlacementDistance
+        {
+            get => defaultPlacementDistance;
+            set
+            {
+                if (defaultPlacementDistance != value)
+                {
+                    defaultPlacementDistance = value;
+                }
+            }
```

Comment:
don't worry about the if not a new value check here, we only need that if we could potentially be doing something expensive

Reply:
Removed